### ISSUE TITLE
[Feature Suggestion] Option to locate missing projects

### ISSUE THREAD

**Issue by etherealxx** [eumario/godot-manager#88]
**Is your feature request related to a problem? Please describe.**
I have several harddisk and love (and need) to swap them once in a while for various reason. When my drive changed the letter placement however, all of the project will appears as missing.
![image](https://github.com/eumario/godot-manager/assets/64251396/ac196716-020a-4a29-8156-5ec5ae10de26)

**Describe the solution you'd like**
Would be nice if there is a Locate function for missing project, just like in Adobe Premiere Pro if you ever used one. When you right click a missing project and pressed locate, you will be presented by a file choosing window. You can search for a substitute folder. When a substitute is chosen, it will compare the metadata (like is the folder name is the same, the project.godot file, etc.) Once a single file is located, the software will also attempt to recover the path of every single project on that folder's root.

For example, let's say there are 2 missing project, F:\gproject\godot1 and F:\gproject\godot2. Both are now located in G:\newproject\*. once you locate the godot1 project (now is located in G:\newproject\godot1, it will attempt to compare the folders inside F:\gproject and G:\newproject, and tries to re-locate all of them.

**Comment by eumario** [comment]
I have implemented this feature.  I've added a new Icon that will appear in the Action Area, when there are missing projects, that will let you browse to the folder that they are located in, and will update the database, with the ones it finds.  Any Missing Projects it doesn't find, will remain in a Missing Status, allowing you to update multiple projects, in multiple folders.

Screenshots:
Missing Projects, showing the new Yellow Scan Folder Icon
![Screenshot 2024-04-02 225917](https://github.com/eumario/godot-manager/assets/63555/86e72348-4748-4142-b231-54ec5d2d8846)
When pressed, a Browse to Folder Dialog popups.
![Screenshot 2024-04-02 225926](https://github.com/eumario/godot-manager/assets/63555/3b20c448-ab26-4628-8efe-0fb92f2bdf08)
When you located where the projects are now located, you Select the folder.
![Screenshot 2024-04-02 225936](https://github.com/eumario/godot-manager/assets/63555/3bda70bb-8b39-4e07-85fb-70b4a1f2d6d1)
Godot will Scan through the folder, and find the missing projects, and update your list of Projects with the new information.
![Screenshot 2024-04-02 225945](https://github.com/eumario/godot-manager/assets/63555/8f1ce436-b01c-45f5-b241-c3b0529cf875)

Let me know if that looks good, I will be pushing an update for Godot Manager in the next day or so.
